internal/helm/controller/controller.go

```diff
@@ -93,6 +104,21 @@ func Add(mgr manager.Manager, options WatchOptions) error {
 	return nil
 }
 
+// parsePredicateSelector parses the selector in the WatchOptions and creates a predicate
+// that is used to filter resources based on the specified selector
+func parsePredicateSelector(selector metav1.LabelSelector) ctrlpredicate.Predicate {
```

Comment:
This function should return an error since the user will not want their operator to run if their selectors are not applied.
```suggestion
func parsePredicateSelector(selector metav1.LabelSelector) (ctrlpredicate.Predicate, error) {
```